### PR TITLE
Fixed #34941 -- Fixed fetching decimal db_default for DecimalFields on Oracle.

### PR DESCRIPTION
**Pull request by felixxm**
This is a long standing bug in bc91f27a86090b4c688b56cd4e37f95eebe6e969 that began manifesting in 7414704e88d73dafbcfbb85f9bc54cb6111439d3.

ticket-34941

### CODE REVIEW

**Comment by felixxm** [comment]
buildbot, test on oracle.

**Comment by pauloxnet** [comment]
I would add a reference to "Oracle" in the title, to make it clear that it is the fix for a bug relating only to that database.

File: django/db/backends/oracle/utils.py [hunk 1]
```diff
@@ -32,7 +33,9 @@ def __init__(self, field):
         self.bound_param = None
 
     def bind_parameter(self, cursor):
-        self.bound_param = cursor.cursor.var(self.db_type)
+        self.bound_param = cursor.cursor.var(
+            self.db_type, arraysize=cursor.cursor.arraysize
```
Comment (nessita) [inline django/db/backends/oracle/utils.py:37]: Do you think is worth to add some sort of comment here explaining why `arraysize` is defined? Following the docs

> The arraysize parameter specifies the number of elements the variable will have. If not specified the bind array size (usually 1) is used. When a variable is created in an output type handler this parameter should be set to the cursor’s array size.

perhaps we could add something like?

`arraysize should be set the cursor’s array size to allow for proper number of elements in the variable created by the output type handler`

**Review by nessita — APPROVED** [review]
Looks great, thank you!

File: django/db/backends/oracle/utils.py (hunk 1 above)
Comment (felixxm) [reply django/db/backends/oracle/utils.py:37]: It's not created in an output type handler, so we can remove it.

**Comment by felixxm** [comment]
buildbot, test on oracle.

File: django/db/backends/oracle/utils.py [hunk 2]
```diff
@@ -23,7 +24,7 @@ class InsertVar:
         "FloatField": Database.NATIVE_FLOAT,
         "DateTimeField": Database.TIMESTAMP,
         "DateField": Database.Date,
```
Comment (charettes) [inline django/db/backends/oracle/utils.py:26]: Do you know if any of these remaining three are also necessary? Does it work on `oracledb` but not `cx_Oracle`?

Comment (felixxm) [reply django/db/backends/oracle/utils.py:26]: > Do you know if any of these remaining three are also necessary?

Yes, they are. You can find corresponding tests in `queries.test_db_returning.ReturningValuesTests`.

> Does it work on `oracledb` but not `cx_Oracle`?

It doesn't work on both :smile:

Comment (felixxm) [reply django/db/backends/oracle/utils.py:26]: ... and I there is ticket-30170 assigned to myself :see_no_evil:

**Comment by felixxm** [comment]
Thanks y'all for checking :+1:

**Pull request merged**